diff --git a/lib/Session.cpp b/lib/Session.cpp
index a0056c1..a5cf58c 100644
--- a/lib/Session.cpp
+++ b/lib/Session.cpp
@@ -199,6 +199,8 @@ void Session::addView(TerminalDisplay * widget)
     QObject::connect( widget ,SIGNAL(changedContentSizeSignal(int,int)),this,
                       SLOT(onViewSizeChange(int,int)));
 
+    updateTerminalSize();
+
     QObject::connect( widget ,SIGNAL(destroyed(QObject *)) , this ,
                       SLOT(viewDestroyed(QObject *)) );
 //slot for close
